jbmc/src/java_bytecode/select_pointer_type.h

```diff
@@ -52,6 +52,9 @@ class select_pointer_typet
       &generic_parameter_specialization_map,
     const namespacet &ns) const;
 
+  /// Get alternative types for a method parameter. For the base class we
```

Comment:
Could you explain what "alternative types" are? There is one place in `java_bytecode/java_entry_point.cpp` where this is used with the comment
```
    // Generate code to allocate and non-deterministicaly initialize the
    // argument, if the argument has different possible object types (e.g., from
    // casts in the function body), then choose one in a non-deterministic way.
```
Maybe that comment could be moved into this doxygen (so it wouldn't be necessary above the call to it any more)?

Reply:
I updated this doc to say that the alternative types are for example the ones appearing in casts but I left the comment in `java_bytecode/java_entry_point.cpp` as it explains how these alternative types are then used in CPROVER_start.